packages/core/src/components/ui/Newsletter/Newsletter.tsx

```diff
@@ -0,0 +1,194 @@
+  /**
+   * The subscribe button label.
+   */
+  subscribeButtonLabel?: string
+  /**
+   * The card Variant
+   */
+  card: Boolean
+}
+
+const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
+  function Newsletter(
+    {
+      icon,
+      title,
+      description,
+      privacyPolicy,
+      emailInputLabel,
+      displayNameInput,
+      nameInputLabel,
+      subscribeButtonLabel,
+      card,
+      ...otherProps
+    },
+    ref
+  ) {
+    // Workaround for dangerouslySetInnerHTML console warning/error.
+    const [render, setRender] = useState(false)
+    useEffect(() => {
+      setRender(true)
+    }, [])
+
+    const { subscribeUser, loading, data } = useNewsletter()
+    const nameInputRef = useRef<HTMLInputElement>(null)
+    const emailInputRef = useRef<HTMLInputElement>(null)
+
+    const { pushToast } = useUI()
+
+    const handleSubmit = (event: FormEvent) => {
+      event.preventDefault()
+      subscribeUser({
+        data: {
+          name: nameInputRef.current?.value ?? '',
+          email: emailInputRef.current?.value ?? '',
+        },
+      })
+
+      if (data?.subscribeToNewsletter?.id) {
+        pushToast({
+          title: 'Hooray!',
+          message: 'Thank for your subscription.',
+          status: 'INFO',
+          icon: <Icon name="CircleWavyCheck" width={30} height={30} />,
+        })
+      } else {
+        pushToast({
+          title: 'Oops.',
+          message: 'Something went wrong. Please Try again.',
+          status: 'ERROR',
+          icon: <Icon name="CircleWavyWarning" width={30} height={30} />,
+        })
+      }
+
+      const formElement = event.currentTarget as HTMLFormElement
+
+      formElement.reset()
+    }
+
+    return (
+      <div data-fs-newsletter={card ? 'card' : ''}>
+        <form
+          data-fs-newsletter-form
+          ref={ref}
+          onSubmit={handleSubmit}
+          {...otherProps}
+          className="layout__content"
+        >
+          <header data-fs-newsletter-header>
+            <h3>
+              <Icon name={icon.icon} width={32} height={32} />
+              {title}
+            </h3>
+            {description && <span> {description}</span>}
+          </header>
+
+          <div data-fs-newsletter-controls>
+            <>
+              <UIInputField
+                inputRef={emailInputRef}
+                id="newsletter-email"
+                label={emailInputLabel}
+                type="email"
+                required
+              />
+              <span
+                data-fs-newsletter-addendum
+                dangerouslySetInnerHTML={{
+                  __html: render && cmsToHtml(privacyPolicy),
+                }}
+              ></span>
+              {displayNameInput ? (
+                <UIInputField
+                  inputRef={nameInputRef}
+                  id="newsletter-name"
+                  label={nameInputLabel}
+                  required
+                />
+              ) : null}
```

Comment:
We need to change the order, it is rendering like this
![Screenshot 2023-05-31 at 10 28 39](https://github.com/vtex/faststore/assets/51174217/7b62b161-3ca5-452d-a030-8d81019a3391)